api/error_unexpected_response_test.go

```diff
@@ -0,0 +1,268 @@
+package api_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"testing/iotest"
+	"time"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/hashicorp/nomad/api"
+	"github.com/shoenig/test/must"
+	"github.com/shoenig/test/portal"
+)
+
+const mockNamespaceBody = `{"Capabilities":null,"CreateIndex":1,"Description":"Default shared namespace","Hash":"C7UbjDwBK0dK8wQq7Izg7SJIzaV+lIo2X7wRtzY3pSw=","Meta":null,"ModifyIndex":1,"Name":"default","Quota":""}`
+
+func TestUnexpectedResponseError(t *testing.T) {
+	t.Parallel()
```

Comment:
the `testutil.Parallel` helper gives us a convenient hook for manipulating parallel behavior - please use it so that we can control these tests down the line if we need to 

(same for all uses of `t.Parallel`)